articles/app-service/overview-managed-identity.md

```diff
@@ -428,11 +428,15 @@ $accessToken = $tokenResponse.access_token
 
 ### <a name="asal"></a>Using the Microsoft.Azure.Services.AppAuthentication library for .NET
 
-For .NET applications and functions, the simplest way to work with a managed identity is through the Microsoft.Azure.Services.AppAuthentication package. This library will also allow you to test your code locally on your development machine, using your user account from Visual Studio, the [Azure CLI](/cli/azure), or Active Directory Integrated Authentication. For more on local development options with this library, see the [Microsoft.Azure.Services.AppAuthentication reference]. This section shows you how to get started with the library in your code.
+For .NET applications and functions, the simplest way to work with a managed identity is through the Microsoft.Azure.Services.AppAuthentication package. This library will also allow you to test your code locally on your development machine, using your user account from Visual Studio, the [Azure CLI](/cli/azure), or Active Directory Integrated Authentication. For more on local development options with this library, see the [Microsoft.Azure.Services.AppAuthentication reference]. 
+
+It's also important to understand that there are subtle differences in the way you code your application to use this library. For instance, if you plan to use the system-assigned managed identity then you can use the code below. However, if you intend to use user-defined managed identities, you need to explicitly configure the AzureServiceTokenProvider with the ClientId of the managed identity. 
+
+This section shows you how to get started with the library in your code.
 
 1. Add references to the [Microsoft.Azure.Services.AppAuthentication](https://www.nuget.org/packages/Microsoft.Azure.Services.AppAuthentication) and any other necessary NuGet packages to your application. The below example also uses [Microsoft.Azure.KeyVault](https://www.nuget.org/packages/Microsoft.Azure.KeyVault).
 
-2. Add the following code to your application, modifying to target the correct resource. This example shows two ways to work with Azure Key Vault:
+2. Add the following code to your application, modifying to target the correct resource, if you plan to run the application locally or using the system-assigned managed identity. This example shows two ways to work with Azure Key Vault:
```

Comment:
This code is valid for all cases - a connection string can be provided via an environment variable to make adjustments.

Reply:
This thread can be resolved. I have created two suggestion comments capturing the change.